### PR TITLE
Support custom partitions

### PR DESCRIPTION
**Pull request by schaefi**
In addition to the volume volume management settings also
allow to setup low level table entries like in the following
example:

```xml
<partitions>
    <partition name="var" size="100" mountpoint="/var" filesystem="ext3"/>
</partitions>
```

### CODE REVIEW

File: doc/source/image_description/elements.rst [hunk 1]
```diff
@@ -707,8 +707,22 @@ that is being used as a vagrant box. For details see: :ref:`setup_vagrant`
...
+contains the root filesystem. Volumes are either a feature
+of the used filesystem or LVM is used for this purpose.
+For details see: :ref:`custom_volumes`
+
+.. note::
+
+   If there is a `<partitions>` section in addition to the
+   `<systemdisk>`, the elements of the `<partitions>` section
+   creates independent slices of the disk and are no longer
+   available in the volumes setup.
```
[the remark this answers is not in the document]
Comment (schaefi) [reply doc/source/image_description/elements.rst]: @isbm ok as a summary what you said makes perfect sense. I'll add that part

[the remark this answers is not in the document]
Comment (schaefi) [reply doc/source/image_description/elements.rst]: done

**Comment by schaefi** [comment]
Here is the integration test to build test custom partitions

* https://build.opensuse.org/package/show/Virtualization:Appliances:Images:Testing_x86:tumbleweed/test-image-partitions-and-volumes

**Comment by schaefi** [comment]
I realized one missing coding part. The calculation of the overal required disk size is currently missing the size constraints from the ```<partitions>```. I'll add that part

@davidcassany Once I have done the size calculations better I'd like to merge this one. Do you see any blockers ?

Thanks much

**Review by davidcassany — APPROVED** [review]
All good from my side :+1:

**Comment by isbm** [comment]
@MofX here, you have it. :wink:

**Pull request merged**